cupyx/scipy/interpolate/_interpolate.py

```diff
@@ -0,0 +1,972 @@
+        out[idx] = CUDART_NAN;
+        return;
+    }
+
+    T vtot = 0;
+    T vb;
+    T va;
+    for(int interval = start_interval; interval <= end_interval; interval++) {
+        const double breakpoint = *&breakpoints[interval];
+        if(interval == end_interval) {
+            vb = eval_poly_1<T>(
+                b - breakpoint, coef, interval, idx, -1, c_dims,
+                stride_0, stride_1);
+        } else {
+            const double next_breakpoint = *&breakpoints[interval + 1];
+            vb = eval_poly_1<T>(
+                next_breakpoint - breakpoint, coef, interval,
+                idx, -1, c_dims, stride_0, stride_1);
+        }
+
+        if(interval == start_interval) {
+            va = eval_poly_1<T>(
+                a - breakpoint, coef, interval, idx, -1, c_dims,
+                stride_0, stride_1);
+        } else {
+            va = eval_poly_1<T>(
+                0, coef, interval, idx, -1, c_dims,
+                stride_0, stride_1);
+        }
+
+        vtot += (vb - va);
+    }
+
+    if(!asc) {
+        vtot = -vtot;
+    }
+
+    out[idx] = vtot;
+
+}
+"""
+
+PPOLY_MODULE = cupy.RawModule(
+    code=PPOLY_KERNEL, options=('-std=c++11',),
+    name_expressions=(
+        [f'eval_ppoly<{type_name}>' for type_name in TYPES] +
+        [f'fix_continuity<{type_name}>' for type_name in TYPES] +
+        [f'integrate<{type_name}>' for type_name in TYPES]))
+
+
+def _get_module_func(module, func_name, *template_args):
+    def _get_typename(dtype):
+        typename = get_typename(dtype)
+        if dtype.kind == 'c':
+            typename = 'thrust::' + typename
+        return typename
+    args_dtypes = [_get_typename(arg.dtype) for arg in template_args]
+    template = ', '.join(args_dtypes)
+    kernel_name = f'{func_name}<{template}>' if template_args else func_name
+    kernel = module.get_function(kernel_name)
+    return kernel
+
+
+def _ppoly_evaluate(c, x, xp, dx, extrapolate, out):
+    """
+    Evaluate a piecewise polynomial.
+
+    Parameters
+    ----------
+    c : ndarray, shape (k, m, n)
+        Coefficients local polynomials of order `k-1` in `m` intervals.
+        There are `n` polynomials in each interval.
+        Coefficient of highest order-term comes first.
+    x : ndarray, shape (m+1,)
+        Breakpoints of polynomials.
+    xp : ndarray, shape (r,)
+        Points to evaluate the piecewise polynomial at.
+    dx : int
+        Order of derivative to evaluate.  The derivative is evaluated
+        piecewise and may have discontinuities.
+    extrapolate : bint
+        Whether to extrapolate to out-of-bounds points based on first
+        and last intervals, or to return NaNs.
+    out : ndarray, shape (r, n)
+        Value of each polynomial at each of the input points.
+        This argument is modified in-place.
+    """
+    # Determine if the breakpoints are in ascending order or descending one
+    ascending = (x[x.shape[0] - 1] >= x[0]).item()
```

Comment:
We don't need `item()` and transfer the value to CPU as `asending` will be used on GPU in `interval_kernel`.

Reply:
I tried it, but it always was false

Reply:
You mean when you passed `ascending` to `find_breakpoint_position ` kernel without using `item()`, the value of `asc` the kernel read was always false?

Reply:
Yes, that was the behavior I saw

Reply:
Thanks. I've also tried to see if we could eliminate the D2H sync. Would you try the following patch to check if it works for you?
https://gist.github.com/takagi/bfd7b5ef66ad69af617ed1ceb9ec3c0f

Reply:
Thanks for the patch, I've just tested it, and it works locally (the tests pass)